src/extension/VSProject.cs

```diff
@@ -348,9 +348,13 @@ private void LoadMSBuildProject()
             MsBuildDocument = _doc;
 
             XmlElement assemblyNameElement = (XmlElement)_doc.SelectSingleNode("/msbuild:Project/msbuild:PropertyGroup/msbuild:AssemblyName", namespaceManager);
+            if (assemblyNameElement == null)
+                throw new ApplicationException("Missing required element AssemblyName");
             string assemblyName = assemblyNameElement.InnerText;
 
             XmlElement outputTypeElement = (XmlElement)_doc.SelectSingleNode("/msbuild:Project/msbuild:PropertyGroup/msbuild:OutputType", namespaceManager);
+            if (outputTypeElement == null)
+                throw new ApplicationException("Missing required element OutputType");
```

Comment:
I wondered why ApplicationException - but it follows convention in the rest of this file. This exception is also caught and converted to an Argument exception in the next file up.